weather/www/js/app.js

```diff
@@ -0,0 +1,147 @@
+var app = angular.module('weather', ['ionic']);
+
+
+app.controller('HomeCtrl', function($scope, weatherService) {
+
+  
+
+  weatherService.get().then(function(response) {
+
+
+//got this idea from stack overflow but couldn't find the link
+    var monthNames = [
+      "January", "February", "March",
+      "April", "May", "June", "July",
+      "August", "September", "October",
+      "November", "December"
+    ];
+
+    var days = [
+      'Sunday', 'Monday', 'Tuesday', 'Wednesday',
+      'Thursday', 'Friday', 'Saturday'
+    ];
+
+
+
+    //w3school website
+    var currentTime = new Date();
+    var hour = currentTime.getHours();
+    var numDay = currentTime.getDate();
+    var day = currentTime.getDay();
+    var monthIndex = currentTime.getMonth();
+    var fullDate = (numDay + ", " + monthNames[monthIndex]);
+
+    console.log(day);
+
+    function time(x) {
+      var newTime = x + hour;
+      if (newTime > 12) {
+        newTime = newTime - 12;
+        if (newTime > 12) {
+          newTime = newTime - 12;
+          return (newTime) + "am";
+        } else {
+          return (newTime) + "pm";
+        }
+
+      } else {
+        return (newTime) + "am";
+      }
+
+    }
+    // got the idea from stack overflow: http://stackoverflow.com/questions/14657378/how-to-set-x-number-of-days-in-the-future-using-javascript
+
+    function dayOfTheWeek(x) {
+      var addDay = x + day;
+      if (addDay > 6) {
+        addDay = addDay - 6;
+      }
+      var newDay = days[addDay];
+      return (newDay);
+    }
+    console.log(dayOfTheWeek(6));
+
+
+    $scope.currentDate = fullDate;
+
+
+    $scope.weather = response.data;
+    console.log($scope.weather);
+    $scope.icon = $scope.weather.currently.icon;
```

Comment:
`` $scope.weather = response.data`` makes everything that is in the @scope.weather object available to your view (index.html). So you don't need to define scope.icon/scope.currentTemp/scope.humidity etc... in your javascript. Once you have:
``$scope.weather = response.data `` 
in your js, you can then use 
`` {{ weather.currently.icon }} ``
in your index.html to get the icon.